app/rtc_s1.py

```diff
@@ -0,0 +1,1262 @@
+    sys.stderr = Logger(logger, logging.ERROR, prefix='[StdErr] ')
+
+    return logger
+
+
+def _update_mosaic_boundaries(mosaic_geogrid_dict, geogrid):
+    xf = geogrid.start_x + geogrid.spacing_x * geogrid.width
+    yf = geogrid.start_y + geogrid.spacing_y * geogrid.length
+    if ('x0' not in mosaic_geogrid_dict.keys() or
+            geogrid.start_x < mosaic_geogrid_dict['x0']):
+        mosaic_geogrid_dict['x0'] = geogrid.start_x
+    if ('xf' not in mosaic_geogrid_dict.keys() or
+            xf > mosaic_geogrid_dict['xf']):
+        mosaic_geogrid_dict['xf'] = xf
+    if ('y0' not in mosaic_geogrid_dict.keys() or
+            geogrid.start_y > mosaic_geogrid_dict['y0']):
+        mosaic_geogrid_dict['y0'] = geogrid.start_y
+    if ('yf' not in mosaic_geogrid_dict.keys() or
+            yf < mosaic_geogrid_dict['yf']):
+        mosaic_geogrid_dict['yf'] = yf
+    if 'dx' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['dx'] = geogrid.spacing_x
+    else:
+        assert(mosaic_geogrid_dict['dx'] == geogrid.spacing_x)
+    if 'dy' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['dy'] = geogrid.spacing_y
+    else:
+        assert(mosaic_geogrid_dict['dy'] == geogrid.spacing_y)
+    if 'epsg' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['epsg'] = geogrid.epsg
+    else:
+        assert(mosaic_geogrid_dict['epsg'] == geogrid.epsg)
+
+
+def _get_raster(output_dir, ds_name, ds_hdf5, dtype, shape,
+                radar_grid_file_dict, output_obj_list, flag_save_vector_1,
+                extension):
+    if flag_save_vector_1 is not True:
+        return None
+
+    output_file = os.path.join(output_dir, ds_name) + '.' + extension
+    raster_obj = isce3.io.Raster(
+        output_file,
+        shape[2],
+        shape[1],
+        shape[0],
+        dtype,
+        "GTiff")
+    output_obj_list.append(raster_obj)
+    radar_grid_file_dict[ds_hdf5] = output_file
+    return raster_obj
+
+
+def _add_output_to_output_metadata_dict(flag, key, output_dir,
+        output_metadata_dict, product_id, extension):
+    if not flag:
+        return
+    output_image_list = []
+    output_metadata_dict[key] = \
+        [os.path.join(output_dir, f'{product_id}_{key}.{extension}'),
+                      output_image_list]
+
+
+def apply_slc_corrections(burst_in: Sentinel1BurstSlc,
+                          path_slc_vrt: str,
+                          path_slc_out: str,
+                          flag_output_complex: bool = False,
+                          flag_thermal_correction: bool = True,
+                          flag_apply_abs_rad_correction: bool = True):
+    '''Apply thermal correction stored in burst_in. Save the corrected signal
+    back to ENVI format. Preserves the phase.'''
+
+    # Load the SLC of the burst
+    burst_in.slc_to_vrt_file(path_slc_vrt)
+    raster_slc_from = gdal.Open(path_slc_vrt)
+    arr_slc_from = raster_slc_from.ReadAsArray()
```

Comment:
```suggestion
    arr_slc_from = raster_slc.ReadAsArray()
```

Reply:
If you agree with removing "from", then please go ahead and correct the names in few following lines